src/core/meshcop/tcat_agent.cpp

```diff
@@ -501,6 +505,93 @@ Error TcatAgent::HandleStartThreadInterface(void)
     return error;
 }
 
+void SeralizeTcatAdvertisementTlv(uint8_t                 *aBuffer,
+                                  uint16_t                &aOffset,
+                                  TcatAdvertisementTlvType aType,
+                                  uint16_t                 aLength,
+                                  const uint8_t           *aValue)
+{
+    aBuffer[aOffset++] = (uint8_t)(aType << 4 | (aLength & 0xf));
+    memcpy(aBuffer + aOffset, aValue, aLength);
+    aOffset += aLength;
+}
+
+uint8_t *TcatAgent::GetAdvertisementData(uint16_t &aLen)
+{
+    aLen = 0;
+
+    if (mVendorInfo == nullptr)
+    {
+        return nullptr;
+    }
+
+    *reinterpret_cast<uint16_t *>(mAdvertisement) = LittleEndian::HostSwap16(OT_TOBLE_SERVICE_UUID);
+    aLen += sizeof(uint16_t);
+    mAdvertisement[2] = OPENTHREAD_CONFIG_THREAD_VERSION << 4 | OT_TCAT_OPCODE;
+    aLen++;
+
+    if (mVendorInfo->mDeviceIds != nullptr)
+    {
+        for (uint8_t i = 0; mVendorInfo->mDeviceIds[i].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY; i++)
+        {
+            switch (MapEnum(mVendorInfo->mDeviceIds[i].mDeviceIdType))
+            {
+            case kTcatDeviceIdOui24:
+                SeralizeTcatAdvertisementTlv(mAdvertisement, aLen, kTlvVendorOui24,
+                                             mVendorInfo->mDeviceIds[i].mDeviceIdLen,
+                                             mVendorInfo->mDeviceIds[i].mDeviceId);
+                break;
+            case kTcatDeviceIdOui36:
+                SeralizeTcatAdvertisementTlv(mAdvertisement, aLen, kTlvVendorOui36,
+                                             mVendorInfo->mDeviceIds[i].mDeviceIdLen,
+                                             mVendorInfo->mDeviceIds[i].mDeviceId);
+                break;
+            case kTcatDeviceIdDiscriminator:
+                SeralizeTcatAdvertisementTlv(mAdvertisement, aLen, kTlvDeviceDiscriminator,
+                                             mVendorInfo->mDeviceIds[i].mDeviceIdLen,
+                                             mVendorInfo->mDeviceIds[i].mDeviceId);
+                break;
+            case kTcatDeviceIdIanaPen:
+                SeralizeTcatAdvertisementTlv(mAdvertisement, aLen, kTlvVendorIanaPen,
+                                             mVendorInfo->mDeviceIds[i].mDeviceIdLen,
+                                             mVendorInfo->mDeviceIds[i].mDeviceId);
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    otBleLinkCapabilities caps = otPlatGetBleLinkCapabilities(&GetInstance());
```

Comment:
Move variable declaration to top of scope? (per style guide).

Reply:
fixed.